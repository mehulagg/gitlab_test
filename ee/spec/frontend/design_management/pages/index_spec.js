import { shallowMount } from '@vue/test-utils';
import Index from 'ee/design_management/pages/index.vue';
import uploadDesignQuery from 'ee/design_management/queries/uploadDesign.graphql';

describe('Design management index page', () => {
  let mutate;
  let vm;

  function createComponent(loading = false) {
    mutate = jest.fn(() => Promise.resolve());
    const $apollo = {
      queries: {
        designs: {
          loading,
        },
        permissions: {
          loading,
        },
      },
      mutate,
    };

    vm = shallowMount(Index, {
      mocks: { $apollo },
      stubs: ['router-view'],
    });

    vm.setData({
      permissions: {
        createDesign: true,
      },
    });
  }

  describe('designs', () => {
    it('renders loading icon', () => {
      createComponent(true);

      expect(vm.element).toMatchSnapshot();
    });

    it('renders error', () => {
      createComponent();

      vm.setData({ error: true });

      expect(vm.element).toMatchSnapshot();
    });

    it('renders empty text', () => {
      createComponent();

      expect(vm.element).toMatchSnapshot();
    });

    it('renders designs list', () => {
      createComponent();

      vm.setData({ designs: ['design'] });

      expect(vm.element).toMatchSnapshot();
    });
  });

  describe('onUploadDesign', () => {
    it('calls apollo mutate', () => {
      createComponent();

      return vm.vm
        .onUploadDesign([
          {
            name: 'test',
          },
        ])
        .then(() => {
          expect(mutate).toHaveBeenCalledWith({
            mutation: uploadDesignQuery,
            variables: {
              files: [{ name: 'test' }],
            },
            optimisticResponse: {
              __typename: 'Mutation',
              uploadDesign: [
                {
                  __typename: 'Design',
                  id: -1,
                  image: '',
                  name: 'test',
                  commentsCount: 0,
                  updatedAt: expect.any(String),
                },
              ],
            },
          });
        });
    });

    it('does not call apollo mutate if createDesign is false', () => {
      createComponent();

      vm.setData({
        permissions: {
          createDesign: false,
        },
      });

      vm.vm.onUploadDesign([]);

      expect(mutate).not.toHaveBeenCalled();
    });

    it('sets isSaving', () => {
      createComponent();

      const uploadDesign = vm.vm.onUploadDesign([
        {
          name: 'test',
        },
      ]);

      expect(vm.vm.isSaving).toBe(true);

      return uploadDesign.then(() => {
        expect(vm.vm.isSaving).toBe(false);
      });
    });
  });
});
