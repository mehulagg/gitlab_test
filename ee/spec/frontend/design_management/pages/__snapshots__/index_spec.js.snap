// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management index page designs does not render toolbar when there is no permission 1`] = `
<div>
  <!---->
   
  <div
    class="mt-4"
  >
    <ol
      class="list-unstyled row"
    >
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          class="design-list-item"
          validateupload="function _default() {
        return undefined;
      }"
        />
      </li>
       
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-1-name"
            id="design-1"
            image="design-1-image"
            notescount="0"
          />
        </design-dropzone-stub>
         
        <!---->
      </li>
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-2-name"
            id="design-2"
            image="design-2-image"
            notescount="1"
          />
        </design-dropzone-stub>
         
        <!---->
      </li>
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-3-name"
            id="design-3"
            image="design-3-image"
            notescount="0"
          />
        </design-dropzone-stub>
         
        <!---->
      </li>
    </ol>
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;

exports[`Design management index page designs renders designs list and header with upload button 1`] = `
<div>
  <header
    class="row-content-block border-top-0 p-2 d-flex"
  >
    <div
      class="d-flex justify-content-between align-items-center w-100"
    >
      <design-version-dropdown-stub />
       
      <div
        class="qa-selector-toolbar d-flex"
      >
        <gl-button-stub
          class="mr-2 js-select-all"
          size="md"
          variant="link"
        >
          Select all
        </gl-button-stub>
         
        <div>
          <delete-button-stub
            buttonclass="btn-danger btn-inverted mr-2"
            buttonvariant=""
          >
            
            Delete selected
            
            <!---->
          </delete-button-stub>
        </div>
         
        <upload-button-stub />
      </div>
    </div>
  </header>
   
  <div
    class="mt-4"
  >
    <ol
      class="list-unstyled row"
    >
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          class="design-list-item"
          validateupload="function _default() {
        return undefined;
      }"
        />
      </li>
       
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-1-name"
            id="design-1"
            image="design-1-image"
            notescount="0"
          />
        </design-dropzone-stub>
         
        <input
          class="design-checkbox"
          type="checkbox"
        />
      </li>
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-2-name"
            id="design-2"
            image="design-2-image"
            notescount="1"
          />
        </design-dropzone-stub>
         
        <input
          class="design-checkbox"
          type="checkbox"
        />
      </li>
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          validateupload="function (_ref2) {
        var dataTransfer = _ref2.dataTransfer;
        var files = Array.from(dataTransfer.files);

        if (files.length > 1) {
          return _error_messages.EXISTING_DESIGN_DROP_MANY_FILES_MESSAGE;
        }

        if (!files.some(function (_ref3) {
          var name = _ref3.name;
          return designFilename === name;
        })) {
          return _error_messages.EXISTING_DESIGN_DROP_INVALID_FILENAME_MESSAGE;
        }

        return false;
      }"
        >
          <design-stub
            event="NONE"
            filename="design-3-name"
            id="design-3"
            image="design-3-image"
            notescount="0"
          />
        </design-dropzone-stub>
         
        <input
          class="design-checkbox"
          type="checkbox"
        />
      </li>
    </ol>
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;

exports[`Design management index page designs renders error 1`] = `
<div>
  <!---->
   
  <div
    class="mt-4"
  >
    <div
      class="alert alert-danger"
    >
      
      An error occurred while loading designs. Please try again.
    
    </div>
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;

exports[`Design management index page designs renders loading icon 1`] = `
<div>
  <!---->
   
  <div
    class="mt-4"
  >
    <gl-loading-icon-stub
      color="orange"
      label="Loading"
      size="md"
    />
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;

exports[`Design management index page when has no designs renders empty text 1`] = `
<div>
  <!---->
   
  <div
    class="mt-4"
  >
    <ol
      class="list-unstyled row"
    >
      <li
        class="col-md-6 col-lg-4 mb-3"
      >
        <design-dropzone-stub
          class="design-list-item"
          validateupload="function _default() {
        return undefined;
      }"
        />
      </li>
       
    </ol>
  </div>
   
  <router-view-stub
    name="default"
  />
</div>
`;
